### PR TITLE
Feature - add drag and drop functionality on import

### PR DESCRIPTION
**Pull request by FraneCaleta**
This feature is about allowing the user to import their previous activites with a drag and drop functionality - previously you could only select it as an attachment.

image

### CODE REVIEW

**Comment by FraneCaleta** [comment]
If merged, this PR should close the issue #1971

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/file-drop.directive.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  @HostListener('drop', ['$event']) onDrop(event: DragEvent) {
+    event.preventDefault();
+    event.stopPropagation();
+
+    // Prevent the browser's default behavior for handling the file drop
+    event.dataTransfer.dropEffect = 'copy';
+
+    const files = event.dataTransfer.files;
+    this.filesDropped.emit(files);
```
Comment (dtslvr) [inline apps/client/src/app/pages/portfolio/activities/import-activities-dialog/file-drop.directive.ts:27]: Just one line?
```
this.filesDropped.emit(event.dataTransfer.files);
```

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.component.ts [hunk 2]
```diff
@@ -175,6 +175,117 @@ export class ImportActivitiesDialog implements OnDestroy {
...
   }
 
+  public onFilesDropped(files: FileList): void {
+    if (files.length === 0) {
+      return;
+    }
+
+    const droppedFile = files[0];
+
+    // Check the file type and handle it accordingly (JSON/CSV)
```
Comment (dtslvr) [inline apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.component.ts:185]: From here, there is a lot of duplicated code. Can you create a function `handleFile(file) {}` with this stuff and reuse it for `onFilesDropped()` and `onSelectFile()`?

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss [hunk 3]
```diff
@@ -32,4 +32,35 @@
...
     top: calc(50% - 10px);
   }
+
+  .custom-button {
+    border: 3px dashed rgba(var(--palette-primary-500), 1);
+    border-radius: 8px;
+    padding: 30px;
+    text-align: center;
+    background-color: inherit;
+    cursor: pointer;
```
Comment (dtslvr) [inline apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss:42]: Can you please use the css class `cursor-pointer` in the html file

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss [hunk 4]
```diff
@@ -32,4 +32,35 @@
     right: 1.5rem;
     top: calc(50% - 10px);
   }
+
+  .custom-button {
+    border: 3px dashed rgba(var(--palette-primary-500), 1);
+    border-radius: 8px;
+    padding: 30px;
```
Comment (dtslvr) [inline apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss:39]: Can you please use the css class `p-4` or `p-5` in the html file

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss [hunk 5]
```diff
@@ -32,4 +32,35 @@
...
+    text-align: center;
```
Comment (dtslvr) [inline apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss:40]: Can you please use the css class `text-center` in the html file

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss [hunk 6]
```diff
@@ -32,4 +32,35 @@
     right: 1.5rem;
     top: calc(50% - 10px);
   }
+
+  .custom-button {
```
Comment (dtslvr) [inline apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss:36]: `.drop-area {`

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss [hunk 7]
```diff
@@ -32,4 +32,35 @@
...
+    border-radius: 8px;
+    padding: 30px;
+    text-align: center;
+    background-color: inherit;
+    cursor: pointer;
+    transition: border-color 0.1s ease;
+    color: rgba(var(--palette-primary-500), 1);
+
+    .button-content {
+      display: flex;
```
Comment (dtslvr) [inline apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss:47]: Can you please use the css class `d-flex` in the html file

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss [hunk 8]
```diff
@@ -32,4 +32,35 @@
...
+      flex-direction: column;
+      align-items: center;
```
Comment (dtslvr) [inline apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss:49]: Can you please use the css class `align-items-center` in the html file

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss [hunk 9]
```diff
@@ -32,4 +32,35 @@
...
+    padding: 30px;
+    text-align: center;
+    background-color: inherit;
+    cursor: pointer;
+    transition: border-color 0.1s ease;
+    color: rgba(var(--palette-primary-500), 1);
+
+    .button-content {
+      display: flex;
+      flex-direction: column;
```
Comment (dtslvr) [inline apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss:48]: Can you please use the css class `flex-column` in the html file

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss [hunk 10]
```diff
@@ -32,4 +32,35 @@
...
+      align-items: center;
+      justify-content: center;
```
Comment (dtslvr) [inline apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss:50]: Can you please use the css class `justify-content-center` in the html file

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.module.ts [hunk 11]
```diff
@@ -13,10 +13,11 @@ import { GfDialogHeaderModule } from '@ghostfolio/client/components/dialog-heade
 import { GfSymbolModule } from '@ghostfolio/client/pipes/symbol/symbol.module';
 import { GfActivitiesTableModule } from '@ghostfolio/ui/activities-table/activities-table.module';
 
+import { FileDropDirective } from './file-drop.directive';
 import { ImportActivitiesDialog } from './import-activities-dialog.component';
 
 @NgModule({
-  declarations: [ImportActivitiesDialog],
+  declarations: [ImportActivitiesDialog, FileDropDirective],
```
Comment (dtslvr) [inline apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.module.ts:20]: Please order alphabetically

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/file-drop.directive.ts [hunk 12]
```diff
@@ -0,0 +1,29 @@
+import { Directive, HostListener, Output, EventEmitter } from '@angular/core';
+
+@Directive({
+  selector: '[appFileDrop]'
```
Comment (dtslvr) [inline apps/client/src/app/pages/portfolio/activities/import-activities-dialog/file-drop.directive.ts:4]: Please rename to `gfFileDrop`

**Review by dtslvr — CHANGES_REQUESTED** [review]
Looking forward to use this feature! 😎 
I have added some comments.

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/file-drop.directive.ts (hunk 1 above)
Comment (FraneCaleta) [reply apps/client/src/app/pages/portfolio/activities/import-activities-dialog/file-drop.directive.ts:27]: Done

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.component.ts (hunk 2 above)
Comment (FraneCaleta) [reply apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.component.ts:185]: Done

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss (hunk 3 above)
Comment (FraneCaleta) [reply apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss:42]: Done

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss (hunk 4 above)
Comment (FraneCaleta) [reply apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss:39]: Done

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss (hunk 5 above)
Comment (FraneCaleta) [reply apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss:40]: Done

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss (hunk 6 above)
Comment (FraneCaleta) [reply apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss:36]: Done. Should I change `.button-content` to `.drop-area-content` then as well?

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss (hunk 7 above)
Comment (FraneCaleta) [reply apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss:47]: Done

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss (hunk 8 above)
Comment (FraneCaleta) [reply apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss:49]: Done

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss (hunk 9 above)
Comment (FraneCaleta) [reply apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss:48]: Done

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss (hunk 10 above)
Comment (FraneCaleta) [reply apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss:50]: Done

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.module.ts (hunk 11 above)
Comment (FraneCaleta) [reply apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.module.ts:20]: Done

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/file-drop.directive.ts (hunk 12 above)
Comment (FraneCaleta) [reply apps/client/src/app/pages/portfolio/activities/import-activities-dialog/file-drop.directive.ts:4]: Done, but what does "gf" stand for? 🤔

Comment (dtslvr) [reply apps/client/src/app/pages/portfolio/activities/import-activities-dialog/file-drop.directive.ts]: Ghostfolio 👻

File: apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss (hunk 6 above)
Comment (FraneCaleta) [reply apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.scss]: @dtslvr Looks like you missed this one :)

**Comment by FraneCaleta** [comment]
> Thank you @FraneCaleta and congrats for your first PR! 🙌🏻
> 
> I did some further improvements...

You're welcome! Thank you for your support throughout the implementation 😄

**Pull request merged**